### PR TITLE
Pip 1506 upgrade cromwell 52 to 59, upgrade autouri to fix redundant md5sum hashing

### PR DESCRIPTION
**Pull request by leepc12**
Multipart download for AWS turns out to be automatically activated.

This PR includes the followings

- Fixed a bug that md5 checksum skipping flag was not propagated for recursive localization (in JSON, CSV, TSV). This is done in [`autouri`'s PR](https://github.com/ENCODE-DCC/autouri/pull/20). So this PR will just upgrade required `autouri` to 0.2.4.
- Wranglers reported random frequent 503 errors from a Caper/Cromwell server. Hard to debug but it's fixed by upgrading Cromwell from 52 to 58 on my side. So upgrade Cromwell.
- While upgrading Cromwell, I found that the old annoying GCP backend's memory-retry feature is fixed and it becomes a global parameter for all backends. It was for GCP backend only at that time. So simply moved memory-retry feature from GCP backend to a global one.
- However memory-retry still doesn't work as intended. So will disable the parameter.
    - Re-opened the related issue on Cromwell's repo. https://github.com/broadinstitute/cromwell/issues/5815
- Checked that metadata DB is compatible between 52 and 58. I was able to use call-caching with 58 from a cache DB generated from 52.
- Updated caper's `CromwellWorkflowMonitor` class to capture workflow/task status change from both Cromwell ==59 and <59.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -5,7 +5,7 @@
 
 setuptools.setup(
     name='caper',
-    version='1.4.2',
+    version='1.5.0',
```
Comment (paul-sud) [inline setup.py:8]: One thought - you could import the version from the caper __init__.py. Then you don't have to change it in two places.

**Review by paul-sud — APPROVED** [review]
Looks good, one comment for future improvement regarding the version numbers.

File: setup.py (hunk 1 above)
Comment (leepc12) [reply setup.py:8]: Sure, I will think about that. I actually tried it last year with `version.py` (like your `__init__.py` method) but had some issues with `pip`. `setup.py` needed to import the package itself to find `__version__` in `__init__.py`.

https://github.com/ENCODE-DCC/caper/blob/a02e27ace6615e7f8681887743dbbd058ed53be3/caper/version.py

Comment (paul-sud) [reply setup.py:8]: Yeah I think you end up having to parse it out of the `__init__.py` without importing the module like you describe. There is some code in `accession`'s `setup.py` to do it: https://github.com/ENCODE-DCC/accession/blob/dev/setup.py#L62

Comment (leepc12) [reply setup.py:8]: Thanks, will take a look and apply that to the next release.

**Pull request merged**